napalm/eos/eos.py

```diff
@@ -166,6 +162,26 @@ def _process_optional_args(self, optional_args):
                 if k in init_args and k not in filter_args
             }
 
+    def _parse_transport(self, transport):
```

Comment:
Could you add some comments pls? This section has become so complex, I'm having a hard time to follow what & how we're trying to accomplish here... 😃

Reply:
Rebased and added comments.  New changeset pushed.